### PR TITLE
Add more CDN options to Backend Buckets

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/7780

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added more fields to cdn_policy block of `google_compute_backend_bucket`
```

### CODE REVIEW

File: products/compute/api.yaml [hunk 1]
```diff
@@ -562,6 +561,64 @@ objects:
...
+        - !ruby/object:Api::Type::Array
+          name: 'negativeCachingPolicy'
+          min_version: beta
+          description: |
+            Sets a cache TTL for the specified HTTP status code. negativeCaching must be enabled to configure negativeCachingPolicy.
+            Omitting the policy and leaving negativeCaching enabled will use Cloud CDN's default cache TTLs.
+          item_type: !ruby/object:Api::Type::NestedObject
+            properties:
+              - !ruby/object:Api::Type::Integer
+                name: 'code'
```
Comment (rileykarson) [inline products/compute/api.yaml:594]: Is this or ttl required? Are neither?

**Review by rileykarson — APPROVED** [review]
Generally looks good, just a quick schema question about the array with a nested schema.

File: products/compute/api.yaml (hunk 1 above)
Comment (upodroid) [reply products/compute/api.yaml:594]: No of the fields are required by the API

**Review by rileykarson — APPROVED** [review]
Hmm- I'm seeing failures on `TestAccComputeBackendBucket_withCdnPolicy` and `TestAccComputeBackendBucket_backendBucketFullExample`. Mind taking a look?

**Comment by rileykarson** [comment]
Whoops, hit the green approve button when I meant to hit comment on that one*

**Comment by upodroid** [comment]
That is weird, `TestAccComputeBackendBucket_backendBucketFullExample` is passing for me. I'll check both out now

**Comment by rileykarson** [comment]
```
------- Stdout: -------
=== RUN   TestAccComputeBackendBucket_backendBucketFullExample
=== PAUSE TestAccComputeBackendBucket_backendBucketFullExample
=== CONT  TestAccComputeBackendBucket_backendBucketFullExample
TestAccComputeBackendBucket_backendBucketFullExample: provider_test.go:266: Step 1/1 error: Error running pre-apply refresh: 2020/11/24 19:30:17 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0
Error: Missing item separator
on config151189280/terraform_plugin_test.tf line 17, in resource "google_compute_backend_bucket" "image_backend":
15:
16:
17:     "X-Tested-By:Magic-Modules"
Expected a comma to mark the beginning of the next item.
--- FAIL: TestAccComputeBackendBucket_backendBucketFullExample (3.64s)
FAIL
```

In case it helps!

**Comment by upodroid** [comment]
TestAccComputeBackendBucket_* tests are passing now.

The CDN Policy test is interesting. 

```
---[ REQUEST ]---------------------------------------
POST /compute/beta/projects/REDACTED/global/backendBuckets?alt=json HTTP/1.1
Host: compute.googleapis.com
User-Agent: Terraform/0.13.2 (+https://www.terraform.io) Terraform-Plugin-SDK/2.1.0 terraform-provider-google-beta/acc
Content-Length: 126
Content-Type: application/json
Accept-Encoding: gzip

{
 "bucketName": "tf-test-ehb4wie18z",
 "cdnPolicy": {
  "signedUrlCacheMaxAgeSec": 1000
 },
 "enableCdn": true,
 "name": "tf-test-psfhiog77x"
}

## SNIP ##

# The post request returned the following:

---[ RESPONSE ]--------------------------------------
HTTP/1.1 200 OK
Connection: close
Transfer-Encoding: chunked
Alt-Svc: h3-29=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Tue, 24 Nov 2020 22:11:50 GMT
Etag: BA9kfIP8kZ-570hqwCt7uioBiXM=/zP_IvhBvR-Ce3iWgyfW1gElYKRM=
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

1e9
{
  "id": "6945798689232272372",
  "creationTimestamp": "2020-11-24T14:11:39.053-08:00",
  "name": "tf-test-psfhiog77x",
  "description": "",
  "selfLink": "https://www.googleapis.com/compute/beta/projects/REDACTED/global/backendBuckets/tf-test-psfhiog77x",
  "bucketName": "tf-test-ehb4wie18z",
  "enableCdn": true,
  "cdnPolicy": {
    "signedUrlCacheMaxAgeSec": "1000",
    "cacheMode": "USE_ORIGIN_HEADERS",
    "negativeCaching": false
  },
  "kind": "compute#backendBucket"
}


```

File: products/compute/terraform.yaml [hunk 2]
```diff
@@ -162,6 +162,10 @@ overrides: !ruby/object:Overrides::ResourceOverrides
           regex: '^(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)$'
       cdnPolicy: !ruby/object:Overrides::Terraform::PropertyOverride
         default_from_api: true
+      cdnPolicy.cacheMode: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
```
Comment (rileykarson) [inline products/compute/terraform.yaml:166]: Could we set explicit defaults instead? Those are generally preferable to `default_from_api`, as they'll show up at plan time.

Comment (upodroid) [reply products/compute/terraform.yaml:166]: Google doesn't set one.

Also, setting signedUrlCacheMaxAgeSec makes Google set cachemode but when you set other fields that field is null. 

Funny API Contract and the docs don't mention the default one either

Comment (rileykarson) [reply products/compute/terraform.yaml:166]: Thanks! `default_from_api` makes sense, in that case.

**Comment by upodroid** [comment]
Test is passing for me.

```
 REDACTED  MCW0CDP3YY  ~  go  …  github.com  hashicorp  terraform-provider-google-beta   master  11⬇  3✎  $    make testacc TEST=./google-beta TESTARGS="-run TestAccComputeBackendBucket_backendBucketFullExample"
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run TestAccComputeBackendBucket_backendBucketFullExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccComputeBackendBucket_backendBucketFullExample
=== PAUSE TestAccComputeBackendBucket_backendBucketFullExample
=== CONT  TestAccComputeBackendBucket_backendBucketFullExample
--- PASS: TestAccComputeBackendBucket_backendBucketFullExample (52.96s)
PASS
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta 54.003s
```

**Review by rileykarson — APPROVED** [review]
LGTM.

The failure in the prior message is expected- it failed because it wasn't recorded in VCR! The test run linked hits the real API, where it passed.

**Pull request merged**